core/templates/pages/learner-dashboard-page/new-home-tab.component.ts

```diff
@@ -0,0 +1,263 @@
+          let learnerGroupsTitle =
+          this.storyIdToLearnerGroupsTitleMap.get(storySummary.getId());
+          storyData.learnerGroupTitle = learnerGroupsTitle ?
+          learnerGroupsTitle[0] : '';
+        }
+        if (storyProgress !== 0) {
+          this.storyInProgress.push(storyData);
+        } else {
+          this.storyInRecommended.push(storyData);
+        }
+      }
+    }
+
+    this.noPlaylistActivity = (
+      (this.explorationPlaylist.length === 0) &&
+    (this.collectionPlaylist.length === 0));
+    this.totalLessonsInPlaylist.push(
+      ...this.explorationPlaylist, ...this.collectionPlaylist);
+
+    this.totalExploration.push(
+      ...this.explorationPlaylist, ...this.incompleteExplorationsList);
+    this.storyExplorationProgressList.push(
+      ...this.storyInProgress, ...this.incompleteExplorationsList);
+
+    for (let exp of this.completedExplorationsList) {
+      this.completedExpIds.push(exp.id);
+    }
+    if (this.totalExploration.length !== 0) {
+      let expIds: string[] = [];
+      for (let exp of this.totalExploration) {
+        expIds.push(exp.id);
+      }
+
+      this.readOnlyExplorationBackendApiService.
+        fetchProgressInExplorationsOrChapters(
+          expIds
+        ).then(explorationsProgressSummary => {
+          let explorationsProgress = explorationsProgressSummary;
+
+          if (explorationsProgress.length !== 0) {
+            for (let i = 0; i < expIds.length; i++) {
+              let progress =
+          this.calculateExplorationProgress(explorationsProgress[i], expIds[i]);
+              this.explorationToProgressMap.set(
+                expIds[i], Math.floor(progress));
+            }
+          }
+        });
+    }
+
+    this.windowIsNarrow = this.windowDimensionService.isWindowNarrow();
+    this.directiveSubscriptions.add(
+      this.windowDimensionService.getResizeEvent().subscribe(() => {
+        this.windowIsNarrow = this.windowDimensionService.isWindowNarrow();
+      }));
+  }
+
+  getTileType(
+      tile: LearnerExplorationSummary | CollectionSummary |
+    storySummaryTile): string {
+    if (tile instanceof LearnerExplorationSummary) {
+      return 'exploration';
+    } else if (tile instanceof CollectionSummary) {
+      return 'collection';
+    }
+    return 'story';
+  }
+
+  calculateExplorationProgress(
+      explorationProgress: ChapterProgressSummary, expId: string): number {
+    let totalCheckpoints = explorationProgress.totalCheckpoints;
+    let visitedCheckpoints = explorationProgress.visitedCheckpoints;
+    if (this.completedExpIds.includes(expId)) {
+      return 100;
+    }
+    if (visitedCheckpoints === 1 && totalCheckpoints === 1) {
+      return 0;
+    }
+    let progress = (visitedCheckpoints / totalCheckpoints) * 100;
+    return progress;
+  }
+
+  isNonemptyObject(object: Object): boolean {
+    return Object.keys(object).length !== 0;
+  }
+
+  getClassroomLink(classroomUrlFragment: string): string {
```

Comment:
Prefer URL over Link to maintain consistency in the codebase.

Reply:
I have to use this as a function. Because I am creating a link using Url Interpolation. Can you Please check the function.
Already used (https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/home-tab.component.ts#L100). PTAL

Reply:
I meant using `getClassroomURL` instead of `getClassroomLink`.